test/mpi/test_mpi.cc

```diff
@@ -247,6 +233,38 @@ void UccTestMpi::set_displ_vsizes(std::vector<ucc_test_vsize_flag_t> &_displs_vs
     displs_vsize = _displs_vsize;
 }
 
+#ifdef HAVE_CUDA
+void UccTestMpi::set_cuda_device(test_set_cuda_device_t set_device)
+{
+    MPI_Comm local_comm;
+    int cuda_dev_count;
+    int local_rank;
+    int device_id;
+    MPI_Comm_split_type(MPI_COMM_WORLD, OMPI_COMM_TYPE_NODE, 0, MPI_INFO_NULL,
```

Comment:
plz change OMPI_COMM_TYPE_NODE to MPI_COMM_TYPE_SHARED, and 0 -> comm_world rank. THis way it will be portable across MPIs and not depend on OMPI (UCC is MPI agnostic)

Reply:
Fixed